src/main/java/seedu/address/storage/DrugSearchUtility.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.storage;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+
+/**
+ * For searching through "src/main/resources/storage/datasetForDrugs.csv"
+ * for drugs whose names contain a certain keyword, then
+ * returning pharmacological information about those drugs.
+ */
+
+public class DrugSearchUtility {
+    private static final String DATASET_PATH = "./src/main/resources/storage/datasetForDrugs.csv"; //filepath to drug dataset
+
+    private static ArrayList<String[]> resultsCache = new ArrayList<>(); //cached results of most recent keyword search
+
+    private static String[] tooGeneric = {"capsule","cream","emulsi","gel","hydrochloride",
+            "injection","lotion","ointment","paste","patch","pill","powder",
+            "solution","supposit","syrup","tablet"};  //keywords too generic to search for, as they will match tens of drugs
+
+    /*
+     * For writing to error log.
+     */
+
+    private static FileWriter f = setF();
+    private static PrintWriter errorLog = new PrintWriter(new BufferedWriter(f));
```

Comment:
Error log has already been provided by the program under the eventbus: attach your class to it and write there instead